sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_http_logging_policy.py

```diff
@@ -41,213 +43,303 @@
     from azure.core.pipeline.transport import (  # pylint: disable=no-legacy-azure-core-http-response-import
         HttpRequest as LegacyHttpRequest,
         HttpResponse as LegacyHttpResponse,
-        AsyncHttpResponse as LegacyAsyncHttpResponse
+        AsyncHttpResponse as LegacyAsyncHttpResponse,
     )
-
+    from azure.core.pipeline.transport._base import _HttpResponseBase as LegacySansIOHttpResponse
+    from azure.core.rest._rest_py3 import _HttpResponseBase as SansIOHttpResponse
 
 HTTPRequestType = Union["LegacyHttpRequest", "HttpRequest"]
-HTTPResponseType = Union["LegacyHttpResponse", "HttpResponse", "LegacyAsyncHttpResponse", "AsyncHttpResponse"]
+HTTPResponseType = Union["LegacyHttpResponse", "HttpResponse", "LegacyAsyncHttpResponse",
+"AsyncHttpResponse", "SansIOHttpResponse", "LegacySansIOHttpResponse"]
 
 
 def _format_error(payload: str) -> str:
     output = json.loads(payload)
-    return output['message'].replace("\r", " ")
+    ret_str = "\n\t" + "Code: " + output['code'] + "\n"
+    message = output["message"].replace("\r\n", "\n\t\t").replace(",", ",\n\t\t")
```

Comment:
The two replace instructions can be collapsed into one - this would redcue overhead of this string interpolation

Reply:
I don't believe it would be possible to combine into a single replace. The replace method takes in a string literal and not a pattern, and then replaces instances of that if it exists as a substring with the new one you specified. From my understanding, chaining it would already be the most efficient way to do this, doing this another way would involve using the python regex library and might make it more complex and potentially have more overhead. Though ultimately with the second replace we exchange processing time using an extra one for having a much more human readable diagnostics log, but whatever that overhead was, it was more than made up for with some of the other changes in this PR that optimized logging.